app/ports/models/port.py

```diff
@@ -107,32 +113,31 @@ def load_ports_and_maintainers_table(ports):
                 except KeyError:
                     pass
 
+        @transaction.atomic
         def load_dependencies_table(ports):
-
-            def load_depends(list_of_dependencies, type_of_dependency, port):
+            def load_depends(list_of_dependencies, type_of_dependency, pid):
                 dependency = Dependency()
                 dependencies = []
                 dependency.type = type_of_dependency
-                dependency.port_name = port
+                dependency.port_name_id = pid
                 for depends in list_of_dependencies:
                     try:
-                        dependencies.append(Port.objects.get(name__iexact=depends.rsplit(':', 1)[-1]))
-                    except Port.DoesNotExist:
-                        print("Failed to append {} as a dependency to {}. Not Found.".format(depends.rsplit(':', 1)[-1],
-                                                                                             port.name))
+                        dependencies.append(port_id_map[depends.rsplit(':', 1)[-1]])
```

Comment:
I would make a separate line saying something like
```python
dependency_name = depends.rsplit ...
dependencies.append(port_id_map[dependency_name])
```
to make it more clear what you are doing (one could also make three lines, but that's not needed).

Reply:
Okay, thanks.